<launch>
  <!-- select a mode: how to drive persons. [opt: 0: data replay;  1: social force;  2: manually control with interface keyboard] -->
  <arg name="person_mode" default="1"/>
  <!-- select a mode: how to drive the robot. [opt: 0: controlled by move_base, listen to the topic /cmd_vel;  2: manually control with interface keyboard] -->
  <arg name="robot_mode" default="0"/>

  <!-- <arg name="scene_file" default="$(find pedsim_simulator)/scenes/demo1_static_env.xml"/> -->
  <arg name="scene_file" default="$(find pedsim_simulator)/scenes/graphicTEB/test11_openspace_lobby_multiP.xml"/>



  <!-- set the initial pose of the robot -->
  <arg name="pose_initial_x" default="-5.0"/>
  <arg name="pose_initial_y" default="0.0"/>
  <arg name="pose_initial_theta" default="0.0"/>

  <!-- parameters for simulator -->
  <arg name="default_queue_size" default="10"/>
  <arg name="enable_groups" default="true"/>
  <arg name="max_robot_speed" default="1.0"/>
  <arg name="simulation_factor" default="1"/>
  <arg name="update_rate" default="25.0"/>
  <arg name="spawn_period" default="5.0"/>

  <!-- Simulator -->
  <node name="pedsim_simulator" pkg="pedsim_simulator" type="pedsim_simulator" output="screen">
    <param name="scene_file" value="$(arg scene_file)" type="string"/>
    <param name="default_queue_size" value="$(arg default_queue_size)" type="int"/>
    <param name="max_robot_speed" value="$(arg max_robot_speed)" type="double"/>
    <param name="robot_mode" value="$(arg robot_mode)" type="int"/>
    <param name="enable_groups" value="$(arg enable_groups)" type="bool"/>
    <param name="simulation_factor" value="$(arg simulation_factor)" type="double"/>
    <param name="update_rate" value="$(arg update_rate)" type="double"/>
    <param name="spawn_period" value="$(arg spawn_period)" type="double"/>  
    <param name="person_mode" value="$(arg person_mode)" type="int"/>
  </node>

  <!-- Fake map_sever -->
  <include file="$(find pedsim_map)/launch/pedsim_map.launch"/>

  <!-- Robot controller -->
  <include file="$(find pedsim_simulator)/launch/robot_controller.launch">
    <arg name="robot_mode" value="$(arg robot_mode)"/>
    <arg name="pose_initial_x" value="$(arg pose_initial_x)"/>
    <arg name="pose_initial_y" value="$(arg pose_initial_y)"/>
    <arg name="pose_initial_theta" value="$(arg pose_initial_theta)"/>
  </include>

  <!-- Open interface_teleop if person_mode==2-->
  <include file="$(find pedsim_simulator)/launch/interface_teleop.launch"/>

  <!-- Description for the robot "Water"-->
  <include file="$(find water_description)/launch/water_description.launch"/>

  <!-- Visualizer -->
  <include file="$(find pedsim_visualizer)/launch/visualizer.launch"/>

  <!-- Rviz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pedsim_simulator)/rviz/pedsim_simulator.rviz"/>

</launch>